better type for the options param in quertRenderedFeatures

beharguy opened the pull request:
## Description
Following [quertRenderedFeatures is not typed well #1900](https://github.com/maplibre/maplibre-gl-js/issues/1900), a type definition added to the options param.
To be consist with the type way the param is typed in `queryRenderedFeatures` function in `query_features.ts` I added an inline definition instead of creating a typescript type.
 
## Launch Checklist

<!-- Thanks for the PR! Feel free to add or remove items from the checklist. -->


 - [X] Confirm **your changes do not include backports from Mapbox projects** (unless with compliant license) - if you are not sure about this, please ask!
 - [X] Briefly describe the changes in this PR.
 - [X] Link to related issues.

HarelM commented:
Thanks for taking the time to open this PR!
Please see the comments in the code, the types that you have added are great, but do not cover all the examples that are given in the JSDocs (different type for the first parameter for example and second optional parameter).
I would suggest to try and address all of the issues in this PR if possible.

beharguy commented:
Hey @HarelM,
By making the geometry also of type QueryRenderedFeaturesOptions allows the function to receive only one param with type QueryRenderedFeaturesOptions (that should be options).
I also made all the properties of QueryRenderedFeaturesOptions optional to support all the examples and tests.

beharguy commented:
@HarelM,
Like you mentioned in the issue, the API is not great and 3.X is the place to change the API.
But as far as I see, also in typescript when creating overloading functions, the similar params between the functions should be the first ones (not the situation here).
If the solution I proposed in the commits of the PR is not good enough then I think an API change is needed.

beharguy commented:
@HarelM,
[X] I've renamed the geometry param to geometryOrOptions like you suggested in #1921.
[X] I changed the code that determinate if geometry is options or not.
[X] I removed the comment about flow.

HarelM commented:
Seems like the changes you made broke a test :-) I'm smiling because we have a test suite that is covering a lot of cases :-)
Please fix the code and I'll merge this PR.
Thanks for time invested in this!

HarelM commented:
Forgot to ask, can you please add a changelog item about this? I've seem some upgrade issues that happens sometimes when changing types, it helps if it documented in the changelog.
